saleor/checkout/calculations.py

```diff
@@ -315,7 +359,13 @@ def _calculate_and_add_tax(
         # Call the tax plugins.
         _apply_tax_data_from_plugins(checkout, manager, checkout_info, lines, address)
         # Get the taxes calculated with apps and apply to checkout.
-        tax_data = manager.get_taxes_for_checkout(checkout_info, lines)
+        tax_data = manager.get_taxes_for_checkout(
+            checkout_info, lines, tax_app_identifier
+        )
+        # If taxAppId is not configured we will for now allow to finalize process for
+        # backward compatibility.
+        if tax_data is None and tax_app_identifier is not None:
```

Comment:
Shouldn't we validate the recieved payload from the app? Currently this will pass if received tax_data will be empty list.